src/TestStack.White/InputDevices/IKeyboard.cs

```diff
@@ -1,13 +1,81 @@
+using TestStack.White.UIItems.Actions;
 using TestStack.White.WindowsAPI;
 
 namespace TestStack.White.InputDevices
 {
     public interface IKeyboard
     {
+        /// <summary>
+        /// Enter a string
+        /// </summary>
+        /// <param name="keysToType">String to enter</param>
         void Enter(string keysToType);
+
+        /// <summary>
+        /// Send a strig to a specific <see cref="IActionListener"/>
+        /// </summary>
+        /// <param name="keysToType">String to send</param>
+        /// <param name="actionListener"><see cref="IActionListener"/> to receive the string</param>
+        void Send(string keysToType, IActionListener actionListener);
```

Comment:
These seem like an internal concept really? We could instead remove all `IActionListener` overloads from the interface, then make `AttachedKeyboard`'s constructor take a `IActionListener`.

Then in `UIItem` can use `new ActionListenerKeyboardDecorator(new Keyboard(), actionListener)`. That way the interfaces are not bloated, and `IAttachedKeyboard` can just inherit from `IKeyboard`